app/state.go

```diff
@@ -732,6 +734,55 @@ func NewUpdateStoreState(in io.ReadCloser, update *datastore.UpdateInfo) State {
 	}
 }
 
+func verifyDependencies(depends, provides map[string]interface{}) error {
+	// Generic closure for checking if element is present in slice.
+	elemInSlice := func(elem string, slice []string) bool {
```

Comment:
Suggestion: I am 90% sure I have seen this duplicated elsewhere, and is really annoying that is not in stdlib. How about creating a util once and for all?